mangaki/mangaki/models.py

```diff
@@ -248,6 +248,31 @@ def update_tags(self, anidb_tags):
 
     # Should add an update_creators method similar to update_tags
 
+    def is_nsfw_based_on_tags(self, anidb_tags):
+        # Might want to move this somewhere else ...
+        potentially_nsfw_tags = ['nudity', 'ecchi', 'pantsu', 'breasts', 'sex',
+                                 'large breasts', 'small breasts', 'gigantic breasts',
+                                 'incest', 'pornography', 'shota', 'masturbation',
+                                 'sexual fantasies', 'anal', 'loli']
+
+        if self.nsfw:
+            return
+
+        sum_weight_all_low = sum([infos["weight"] for t, infos in anidb_tags.items()
```

Comment:
As you don't seem to use the `t` variable, you could just use `anidb_tags.values()` which returns only the `infos`:

```python
sum_weight_all_low = sum((infos["weight"] for infos in anidb_tags.values())
```

Also, it's preferable to use generator expression: `(x for x in iterator)` rather than `[x for x in iterator]` when you don't need to put the entire data in the memory.

A list comprehension will take more memory than a generator expression in most of the cases, in this case, `sum` is adapted for generators because it only has to look at one value at a time, on the other hand, if you need to access all the content at the same time, then generators will be a bit more painful than lists (which you can randomly access).